cachi2/core/package_managers/generic.py

```diff
@@ -1,13 +1,54 @@
-from cachi2.core.errors import PackageRejected
+import logging
+from typing import Literal, Optional
+
+import yaml
+from pydantic import BaseModel, ValidationError, field_validator
+
+from cachi2.core.errors import PackageManagerError, PackageRejected
 from cachi2.core.models.input import Request
 from cachi2.core.models.output import RequestOutput
 from cachi2.core.models.sbom import Component
 from cachi2.core.rooted_path import RootedPath
 
+LOG = logging.getLogger(__name__)
 DEFAULT_LOCKFILE_NAME = "cachi2_generic.yaml"
 DEFAULT_DEPS_DIR = "deps/generic"
 
 
+class LockfileMetadata(BaseModel):
+    """Defines format of the metadata section in the lockfile."""
+
+    version: Literal["1.0"]
+
+
+class LockfileArtifact(BaseModel):
+    """Defines format of a single artifact in the lockfile."""
+
+    download_url: str
```

Comment:
Are we postponing adding the `purl` field?

Reply:
Yes, as agreed upon [in the design](https://github.com/containerbuildsystem/cachi2/pull/652#discussion_r1777195032), the purl functionality will be available only for the maven artifacts.